spring-xsuaa/src/main/java/com/sap/xs2/security/container/UserInfo.java

```diff
@@ -630,6 +631,29 @@ private URI replaceSubdomain(URI uri, String subdomain) {
 				.path(uri.getPath());
 		return uri.resolve(builder.build().toString());
 	}
+
+	@Override
+	public String toString() {
+		return getName();
+	}
+
+	@Override
+	public String getName() {
+		String principalName = null;
+		try {
+			if (GRANTTYPE_CLIENTCREDENTIAL.equals(getGrantType())) {
+				principalName = getClientId();
+			} else {
+				principalName = getLogonName();
+				if (getOrigin() != null) {
+					principalName = getOrigin() + "." + principalName;
```

Comment:
The "." is a very common seperator. I'd rather use something like /,# or |

Reply:
@mwdb 
Thanks, Martijn! I've corrected the delimiter to "/"... maybe in future we can make it configurable...